### PR TITLE
[IMP] web: set default time and prevent shifting in the daterange picker

### PR DESCRIPTION
**Pull request by rkka-odoo**
PURPOSE-

- Prevent flickering on clicking apply in daterange picker.
- The time selector in the daterange,  time default to the current hour(rounded down to the hour) and the 'to' time default to the next hour .
 - Stop automatic shifting of month when we select a range in the month on the
 right panel.

SPECIFICATIONS-

- Check the current default hours and increase the hours by +1 if index==1.
- prevent automatic shifting of month in daterange.

Task:3471424

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by aab-odoo** [comment]
runbot is red @sauc-odoo @rkka-odoo

**Comment by sauc-odoo** [comment]
> runbot is red @sauc-odoo @rkka-odoo

Hello @aab-odoo 
The branch is green now. Thanks!!

File: addons/web/static/src/core/datetime/datetime_picker.js [hunk 1]
```diff
@@ -421,8 +421,8 @@ export class DateTimePicker extends Component {
             throw new Error(`DateTimePicker error: given "maxDate" comes before "minDate".`);
         }
 
-        const timeValues = this.values.map((val) => [
-            (val || DateTime.local()).hour,
+        const timeValues = this.values.map((val,index) => [
```
Comment (aab-odoo) [inline addons/web/static/src/core/datetime/datetime_picker.js:424]: lint/prettify your code (you can enable the tooling via /addons/web/tooling/enable.sh)

**Comment by sauc-odoo** [comment]
Hello @Arcasias 
Changes have been done, Please verify it.
Thanks!!

**Review by aab-odoo — COMMENTED** [review]
As far as I can see, this commit fixes 3 issues (issue/imp, whatever), but tests only reflect one of them.

**Review by Arcasias — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**